parse/parse.go

```diff
@@ -39,6 +43,68 @@ func GoList(stdIn *bufio.Scanner) (deps types.ProjectList, err error) {
 	return deps, nil
 }
 
+// GoListAgnostic will take a io.Reader that is likely the os.StdIn, try to parse it as
+// if `go list -json -m all` was ran, and then try to reparse it as if `go list -m all`
+// was ran instead. It returns either an error, or a deps of types.ProjectList
+func GoListAgnostic(stdIn io.Reader) (deps types.ProjectList, err error) {
+	// stdIn should never be massive, so taking this approach over reading from a stream
+	// multiple times
+	johnnyFiveNeedInput, err := ioutil.ReadAll(stdIn)
+	if err != nil {
+		return
+	}
+	decoder := json.NewDecoder(strings.NewReader(string(johnnyFiveNeedInput)))
+
+	for {
+		var mod types.GoListModule
+
+		err = decoder.Decode(&mod)
+		if err == io.EOF {
+			err = nil
+			break
+		}
+		if err != nil {
+			break
+		}
+
+		project, err := modToProjectList(mod)
+		if _, ok := err.(*NoVersionError); ok {
+			continue
+		}
+		deps.Projects = append(deps.Projects, project)
+	}
+
+	if err != nil {
+		err = nil
+		scanner := bufio.NewScanner(strings.NewReader(string(johnnyFiveNeedInput)))
+		deps, err = GoList(scanner)
+		if err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+func modToProjectList(mod types.GoListModule) (dep types.Projects, err error) {
+	if mod.Replace != nil {
+		if mod.Replace.Version == "" {
+			err = &NoVersionError{err: fmt.Errorf("No version found for mod")}
+			return
+		}
+		dep.Name = mod.Replace.Path
+		dep.Version = mod.Replace.Version
+		return
+	}
```

Comment:
Since #142 called this out as an issue i think we should add a test around it to make sure it is working as expected.

Reply:
I have one! https://github.com/sonatype-nexus-community/nancy/pull/150/files#diff-7b16ed199b45ba365ab40918a65bbb2bR64-R78

Reply:
My bad i should have been more clear :) that just handles `replace` for `go list` and not `go list -json`. We should probably have a replace test for the json path too. 

```
{
        "Path": "github.com/gorilla/websocket",
        "Version": "v1.4.0",
        "Replace": {
                "Path": "github.com/gorilla/websocket",
                "Version": "v1.4.2",
                "Time": "2020-03-19T17:50:51Z",
                "GoMod": "/Users/ME/go/pkg/mod/cache/download/github.com/gorilla/websocket/@v/v1.4.2.mod"
        },
        "Update": {
                "Path": "github.com/gorilla/websocket",
                "Version": "v1.4.2",
                "Time": "2020-03-19T17:50:51Z"
        },
        "Indirect": true,
        "GoMod": "/Users/ME/go/pkg/mod/cache/download/github.com/gorilla/websocket/@v/v1.4.2.mod"
}
```

One of those testdata files but with that and maybe another in it.

Reply:
In: 64079e0